source/server/hot_restarting_child.cc

```diff
@@ -32,35 +32,36 @@ int HotRestartingChild::duplicateParentListenSocket(const std::string& address)
   return wrapped_reply->reply().pass_listen_socket().fd();
 }
 
-void HotRestartingChild::getParentStats(HotRestart::GetParentStatsInfo& info) {
-  memset(&info, 0, sizeof(info));
+std::unique_ptr<HotRestartMessage> HotRestartingChild::getParentStats() {
   if (restart_epoch_ == 0 || parent_terminated_) {
-    return;
+    return nullptr;
   }
 
   HotRestartMessage wrapped_request;
   wrapped_request.mutable_request()->mutable_stats();
   sendHotRestartMessage(parent_address_, wrapped_request);
 
   std::unique_ptr<HotRestartMessage> wrapped_reply = receiveHotRestartMessage(Blocking::Yes);
-  RELEASE_ASSERT(replyIsExpectedType(wrapped_reply.get(), HotRestartMessage::Reply::kStats),
-                 "Did not get a StatsReply for our StatsRequest.");
-  // TODO(fredlas) this is where the stat transferring, to be added later in this PR, will go.
-  info.memory_allocated_ = wrapped_reply->reply().stats().memory_allocated();
-  info.num_connections_ = wrapped_reply->reply().stats().num_connections();
+  if (!replyIsExpectedType(wrapped_reply.get(), HotRestartMessage::Reply::kStats)) {
```

Comment:
Can this happen? Should this be a RELEASE_ASSERT?

Reply:
Maybe if the parent somehow crashes (or decides to exit or gets terminated) while doing the hot restart handoff? I don't have enough context to have even an uninformed opinion on whether effort should be made to let the hot restarting child successfully start up in such a case, but looking just at what's going on here, you are only missing out on a round (albeit maybe the first one) of stats updates from the parent.

Reply:
In general we assume the protocol "just works" and crash fast if not. I would probably RELEASE_ASSERT here to keep error handling to a minimum.

Reply:
SGTM, will do.